sys/src/cmd/screenlock.c

```diff
@@ -241,21 +240,23 @@ lockscreen(void)
 	procrfork(blanker, nil, 4096, RFFDG);
 	fd = open(pic, OREAD);
 	if(fd > 0){
+		draw(screen, screen->r, display->black, nil, ZP);
```

Comment:
I would argue that the blank should be even before the open, but outside the if. As it is it looks like your blanking is dependent on the if. I may still be misreading it.